includes/blocks/course-theme/class-quiz-actions.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * File containing the Quiz_Actions class.
+ *
+ * @package sensei
+ * @since 4.0.0
+ */
+
+namespace Sensei\Blocks\Course_Theme;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+use \Sensei_Blocks;
+use Sensei_Quiz;
+
+/**
+ * Class Quiz_Actions is responsible for rendering the quiz
+ * actions button as well as quiz pagination if enabled.
+ */
+class Quiz_Actions {
+	/**
+	 * Quiz_Actions constructor.
+	 */
+	public function __construct() {
+		Sensei_Blocks::register_sensei_block(
+			'sensei-lms/quiz-actions',
+			[
+				'render_callback' => [ $this, 'render' ],
+			]
+		);
+	}
+
+	/**
+	 * Renders the block.
+	 *
+	 * @access private
+	 *
+	 * @return string The block HTML.
+	 */
+	public function render() : string {
+
+		if ( ! sensei_can_user_view_lesson() ) {
+			return '';
+		}
+
+		\Sensei_Quiz::start_quiz_questions_loop();
+		global $sensei_question_loop;
+		$pagination_enabled = $sensei_question_loop['total_pages'] > 1;
+
+		// Get quiz actions. Either actions with pagination
+		// or only action if pagination is not enabled.
+		ob_start();
+		if ( $pagination_enabled ) {
+			Sensei_Quiz::the_quiz_pagination();
+			Sensei_Quiz::output_quiz_hidden_fields();
+		} else {
+			Sensei_Quiz::action_buttons();
+		}
+		$actions = ob_get_clean();
+
+		if ( ! $actions ) {
+			return '';
+		}
+
+		return ( "
+			<form method='POST' enctype='multipart/form-data'>
+				{$actions}
+			</form>
```

Comment:
Just having the `<form>` around the buttons means the submission won't include the actual answers, since those inputs are outside the form. It only works now because there is a form wrapper in the quiz template already. But then the browser will just ignore this form, since it's not valid to nest them.

I've been looking at replacing the form in the quiz template, and think we should just add a global form from filters if needed, instead of it being a dedicated wrapper block that can be removed to break thinks. 

Here is the code, it might be better suited for this PR and in the quiz-actions block:

```php
/**
	 * Add a form wrapper for quiz pages.
	 *
	 * @access private
	 */
	private function maybe_add_form_wrapper() {

		if ( ! has_block( 'sensei-lms/quiz-actions' ) ) {
			return;
		}

		add_action( 'wp_body_open', [ $this, 'start_form' ] );
		add_action( 'wp_footer', [ $this, 'end_form' ] );
	}

	/**
	 * Add a form open tag.
	 *
	 * @access private
	 */
	public function start_form() {
		echo '<form method="post" enctype="multipart/form-data" class="sensei-form">';
	}

	/**
	 * Add a form close tag.
	 *
	 * @access private
	 */
	public function end_form() {
		echo '</form>';
	}
```

Reply:
> Just having the `<form>` around the buttons means the submission won't include the actual answers, since those inputs are outside the form.

Nice catch! Don't know what I was thinking.

> instead of it being a dedicated wrapper block that can be removed to break thinks.

You mean user can potentially remove it in FSE mode?

> I've been looking at replacing the form in the quiz template, and think we should just add a global form from filters if needed,

Hmm. But this way it will wrap everything, including the admin bar. Not sure if that's safe.

Reply:
I just updated the logic for quiz actions. Now it is rendered inside a <form> if it is not already a part of a larger quiz form. 93a028e578b4baa26755249ace50890d162dfb9e

Reply:
Yeah if we rely on a 'Sensei Quiz Form' or similar block to be needed around the content and action blocks, it could become pretty unreliable, (Eg users removing it, or starting a template from scratch in the future).

I'll just handle this then in my PR for moving that top-level form out from the template.

Reply:
Sounds good.